scikits/odes/sundials/cvode.pyx

```diff
@@ -1166,3 +1204,22 @@ cdef class CVODE:
         if not self.y0   is NULL: N_VDestroy(self.y0)
         if not self.y    is NULL: N_VDestroy(self.y)
         if not self.atol is NULL: N_VDestroy(self.atol)
+
+    def validate_flags(self, flag, t_vals, y_vals, t_err, y_err):
+        if flag not in self.RETURN_FLAGS:
+            raise NotImplementedError(self.UNKNOWN_FLAG_MESSAGE.format(flag))
+        elif self.RETURN_FLAGS[flag] == "SUCCESS":
+            return t_vals, y_vals
+        else:
+            error_name = self.RETURN_FLAGS[flag]
+            if flag < 0:
+                error = RuntimeError(ERROR_STR.format(error_name, t_err, y_err))
+                error.error_name = error_name
+                error.t_vals = t_vals
+                error.y_vals = y_vals
+                error.t_err = t_err
+                error.y_err = y_err
+                raise error
+            else:
+                warn(WARNING_STR.format(error_name, t_err, y_err))
```

Comment:
I don't like the handling of TSTOP and ROOT (flags 1, 2). It is needed to know if TSTOP or ROOT was reached. We can't have people check on the warning string to know what happened.